json_object.c

```diff
@@ -1478,7 +1485,7 @@ static int json_object_copy_serializer_data(struct json_object *src, struct json
 	if (dst->_to_json_string == json_object_userdata_to_json_string ||
 	    dst->_to_json_string == _json_object_userdata_to_json_string)
 	{
-		dst->_userdata = strdup(src->_userdata);
+		dst->_userdata = strdup((char*)src->_userdata);
```

Comment:
Is this actually needed?  I really don't want to be littering this code with casts to cater to a broken compiler if we can avoid it.  :(

Reply:
No, you're right, I've managed to get around this one.

Reply:
Now removed from the changes.